### PR TITLE
made admin panel for users and questions

### PR DESCRIPTION
**Pull request by ppauljonathan**
made admin panel for user and questions unpublishing

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -50,6 +50,10 @@ def send_reset_mail
     UserMailer.with(user_id: id).reset_email.deliver_later
   end
 
+  def soft_destroy
```
Comment (shubhamjuneja1) [inline app/models/user.rb:53]: rename to mark_disabled

File: app/models/question.rb [hunk 2]
```diff
@@ -42,6 +42,10 @@ def to_param
     url_slug
   end
 
+  def unpublish
+    update(published_at: nil, save_as_draft: true)
```
Comment (shubhamjuneja1) [inline app/models/question.rb:46]: what is save_as_draft doing here?

File: app/controllers/admin/users_controller.rb [hunk 3]
```diff
@@ -0,0 +1,6 @@
+class Admin::UsersController < ApplicationController
+  before_action :check_admin
```
Comment (shubhamjuneja1) [inline app/controllers/admin/users_controller.rb:2]: all admin controllers should inherit from base_controller in admin which should have this before_action

**Review by shubhamjuneja1 — CHANGES_REQUESTED** [review]

File: app/models/question.rb (hunk 2 above)
Comment (ppauljonathan) [reply app/models/question.rb:43]: moved to concern, changed to `update_columns`, so as to not trigger callbacks

File: app/controllers/admin/users_controller.rb (hunk 3 above)
Comment (ppauljonathan) [reply app/controllers/admin/users_controller.rb:2]: done

**Review by shubhamjuneja1 — CHANGES_REQUESTED** [review]
Changes

**Review by shubhamjuneja1 — APPROVED** [review]

**Pull request merged**